ts/src/bigone.ts

```diff
@@ -577,6 +635,65 @@ export default class bigone extends Exchange {
             };
             result.push (entry);
         }
+        for (let i = 0; i < contractResponse.length; i++) {
+            const market = contractResponse[i];
+            const marketId = this.safeString (market, 'symbol');
+            const baseId = marketId.slice (0, 3);
+            const quoteId = marketId.slice (3);
+            const inverse = (quoteId === 'USD');
+            const settleId = inverse ? baseId : quoteId;
+            const base = this.safeCurrencyCode (baseId);
+            const quote = this.safeCurrencyCode (quoteId);
+            const settle = this.safeCurrencyCode (settleId);
+            result.push ({
```

Comment:
@samgermain we now have a `safeMarketStructure` that we can pass the markets through